app/models/article.rb

```diff
@@ -52,6 +52,10 @@ def published?
     !wip?
   end
 
+  def generate_token_with_secure_random_base64
+    self.token = wip ? (token.presence || SecureRandom.urlsafe_base64) : nil
+  end
```

Comment:
僕の書き方がわかりづらくてすみません。
SecureRandomのbase64であるということは中身を見ればわかるので、メソッド自体は関係な表現がいいと思います。（副作用のあるメソッドなので末尾に!を付けました）

```suggestion
  def generate_token!
    self.token ||= SecureRandom.urlsafe_base64
  end
```

そして、そのメソッドはtoken生成だけに特化して、wipだったらなどは他の部分でやった方がわかりやすいとおもいます。（このメソッドはtokenと生成していて、それが何方式で作られているかを説明するためだけのメソッドにした方がわかりやすいとおもうので）